### ISSUE TITLE
[Feature request] Pylint should report wrong except clause(s)

### ISSUE THREAD

**Issue by TheSnoozer** [PyCQA/pylint#2494]
### Is your feature request related to a problem? Please describe
I recently saw some code that tries to catch multiple exceptions using the `|` operator. As per [python documentation](https://docs.python.org/3/tutorial/errors.html#handling-exceptions) that should be done with a colon (e.g. ```except (RuntimeError, TypeError, NameError)```).

For demonstration consider the following example:
```python
def call_me():
    try:
        raise ValueError("A problem")
    except (ValueError | TypeError) as error:
        print("Error: {0}!!".format(error))


print("hello")
```

Unfortunately pylint doesn't complain about it. Running the code without calling the function works fine, but when calling the function python complains with
```
Original exception was:
Traceback (most recent call last):
  File "example.py", line 3, in call_me
    raise ValueError("A problem")
ValueError: A problem

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "example.py", line 9, in <module>
    call_me()
  File "example.py", line 4, in call_me
    except (ValueError | TypeError) as error:
TypeError: unsupported operand type(s) for |: 'type' and 'type'
```

### Describe the solution you'd like
Even though that might be a rare case and somewhat obvious for most python developers, I think that pylint should report a warning or even an error when someone is using  ```except (ValueError | TypeError) as error:``` instead of ```except (ValueError, TypeError) as error:```

**Comment by PCManticore** [comment]
Could be useful, but not sure right now if it should be a core error or an extension in `pylint.extensions`, as it does seem to be a rare kind of error.
